### PR TITLE
Feature/dynamic stepper

### PR DESCRIPTION
**Pull request by daileytj**
Fixes PXBLUE 1282.

#### Changes proposed in this Pull Request:
- Add dynamic stepper design pattern example
- Update tsconfig files

### CODE REVIEW

**Comment by jeffvg** [comment]
After adding step seeing this in console

Error: ExpressionChangedAfterItHasBeenCheckedError: Expression has changed after it was checked. Previous value for 'tabIndex': '0'. Current value: '-1'.
    Angular 35
    RxJS 5
    Angular 9

File: src/app/pages/dynamic-stepper/dynamic-stepper.component.spec.ts [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+import { BrowserAnimationsModule } from '@angular/platform-browser/animations';
+
+describe('DynamicStepperComponent', () => {
+    let component: DynamicStepperComponent;
+    let fixture: ComponentFixture<DynamicStepperComponent>;
+
+    beforeEach(async(() => {
+        void TestBed.configureTestingModule({
+            imports: [
+                DynamicStepperModule,
```
Comment (emclaug2) [inline src/app/pages/dynamic-stepper/dynamic-stepper.component.spec.ts:21]: You should be able to replace all these individual imports with DynamicStepperModule

File: src/app/pages/dynamic-stepper/dynamic-stepper.component.html [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+                        *ngIf="stepper.selectedIndex === i"
+                        data-cy="delete-step"
+                        (click)="removeStep(i)"
+                        matTooltip="Remove step"
+                        matTooltipPosition="right"
+                        >delete</mat-icon
+                    >
+                </ng-template>
+                <mat-radio-group class="radio-group" (change)="changeStepSelection($event, i)" data-cy="radiogroup">
+                    <mat-radio-button style="margin: 10px 0;" *ngFor="let option of stepOptions" [value]="option.value">
```
Comment (joebochill) [inline src/app/pages/dynamic-stepper/dynamic-stepper.component.html:34]: Can we update all the margins/paddings in here that are using a multiple of 5 (old standard) to use multiples of 8 (material standard)?

Comment (joebochill) [reply src/app/pages/dynamic-stepper/dynamic-stepper.component.html:34]: There might be an instance or two of this in the patterns that are already merged in too

File: src/app/pages/dynamic-stepper/dynamic-stepper.component.html [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+    </mat-vertical-stepper>
+    <div style="padding-left: 20px;">
+        <button mat-raised-button color="primary" data-cy="done" (click)="handleDone()">
+            DONE
+        </button>
+    </div>
+</div>
+<div *ngIf="procedureFinished" style="padding: 20px 30px;" data-cy="reset-page">
+    <p>Procedure created successfully.</p>
+    <button mat-raised-button color="primary" (click)="handleReset()" data-cy="reset">
```
Comment (joebochill) [inline src/app/pages/dynamic-stepper/dynamic-stepper.component.html:72]: Should we use an empty state for this?

File: src/app/pages/dynamic-stepper/dynamic-stepper.component.scss [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+    padding: 16px 64px;
+    background: transparent;
+    text-align: left;
+
+    &[disabled] {
+        background: transparent;
+    }
+}
+
+::ng-deep .mat-step:last-child .mat-step-label {
```
Comment (joebochill) [inline src/app/pages/dynamic-stepper/dynamic-stepper.component.scss:27]: do we need ng deep on these (e.g., if we change the viewEncapsulation)?

File: src/app/pages/dynamic-stepper/dynamic-stepper.component.ts [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+    constructor(
+        private readonly _drawerService: StateService,
+        private readonly _breakpointObserver: BreakpointObserver,
+        private readonly _formBuilder: FormBuilder
+    ) {}
+
+    ngOnInit(): void {
+        this.RandomFormGroup = this._formBuilder.group({
+            Ctrl: ['', Validators.required],
+        });
```
Comment (joebochill) [inline src/app/pages/dynamic-stepper/dynamic-stepper.component.ts:46]: What is this random form group?

**Comment by joebochill** [comment]
Should Dynamic Stepper be a top-level menu item since it's the only thing in the Steppers bucket?

**Comment by emclaug2** [comment]
> Should Dynamic Stepper be a top-level menu item since it's the only thing in the Steppers bucket?

If we plan on adding more steppers then maybe we can get away with it being nested.  Otherwise the loading states is in a single item bucket too.

File: src/app/pages/dynamic-stepper/dynamic-stepper.component.scss (hunk 4 above)
Comment (daileytj) [reply src/app/pages/dynamic-stepper/dynamic-stepper.component.scss:27]: looks like we do need ng-deep here regardless of ViewEncapsulation, unless I can find a more specific selector. Haven't had any luck yet.

File: src/app/pages/dynamic-stepper/dynamic-stepper.component.ts (hunk 5 above)
Comment (daileytj) [reply src/app/pages/dynamic-stepper/dynamic-stepper.component.ts:46]: Removing this...

**Comment by daileytj** [comment]
> > Should Dynamic Stepper be a top-level menu item since it's the only thing in the Steppers bucket?
> 
> If we plan on adding more steppers then maybe we can get away with it being nested. Otherwise the loading states is in a single item bucket too.

Let me know what you guys want me to do here

File: src/app/pages/dynamic-stepper/dynamic-stepper.component.html (hunk 3 above)
Comment (daileytj) [reply src/app/pages/dynamic-stepper/dynamic-stepper.component.html:72]: Will do.

**Comment by joebochill** [comment]
> > > Should Dynamic Stepper be a top-level menu item since it's the only thing in the Steppers bucket?
> > 
> > 
> > If we plan on adding more steppers then maybe we can get away with it being nested. Otherwise the loading states is in a single item bucket too.
> 
> Let me know what you guys want me to do here

I was going to make the same comment on the Loading States example when we move it over. It's already a top-level item in React. And I don't suspect we're going to add any other stepper examples for a while.

File: src/app/pages/dynamic-stepper/dynamic-stepper.component.scss [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+    vertical-align: middle;
+    color: map-get($pxb-black, 500);
+    font-size: 18px;
+    line-height: 24px;
+}
+
+.add-button {
+    width: 100%;
+    height: 100%;
+    padding: 16px 64px;
```
Comment (joebochill) [inline src/app/pages/dynamic-stepper/dynamic-stepper.component.scss:18]: This is going to sound very nitpicky, but the text in the Add a step row and the numbers steps are off by 4 pixels (64 vs 60). If you change this from 64px to 60, they'll line up...it'll be easier than changing the margin on the steps bc i think that's coming from theme styles.

**Pull request merged**